templates/result.html

```diff
@@ -8,9 +8,211 @@
+        <li>
+            <h2 class='subtitle'><a href='{{data.homepage_url}}' target='_blank'>{{data.name}}</a> </h2>
+            <span style="color: gray;"> (H_Index: {{data.h_index}}, Works_count: {{data.works_count}}, Country_code:
+                {{data.country_codes}})<br> {{data.description}} </span>
+        </li>
+
+        <div style="margin-left:10px;">
+            <p></p>
+        </div>
+        {% endif %}
+        {% endif %}
+        {% if title == 'Funder' %}
+        {% if data %}
+        <p class='url'>{{data.url}}</p>
+        <li>
+            <h2 class='subtitle'><a href='{{data.homepage_url}}' target='_blank'>{{data.name}}</a> </h2>
+            <span style="color: gray;"> (Grants: {{data.grants_count}}, Works_count: {{data.works_count}})
+                <br>{{data.description}}</span>
+        </li>
+
+        <div style="margin-left:10px;">
+            <p></p>
+        </div>
+        {% endif %}
+        {% endif %}
+        {% if title == 'Image' %}
+        {% if data %}
+        <p class='url'>{{data.url}}</p>
+        <li>
+            <h2 class='subtitle'><a href='{{data.url}}' target='_blank'>{{data.title}}</a> </h2>
+            <span style="color: gray;"> - {{data.date}}</span>
+        </li>
+
+        <div style="margin-left:10px;">
+            <p>{{data.authors}}</p>
+        </div>
+        {% endif %}
+        {% endif %}
+        {% if title == 'Zenodo' %}
+        {% if data %}
+        <p class='url'>{{data.url}}</p>
+        <li>
+            <h2 class='subtitle'><a href='{{data.url}}' target='_blank'>{{data.title}}</a></h2>
+            <span style="color: gray;"> - {{data.date}}<br>{{data.description}}</span>
+        </li>
+
+        <div style="margin-left:10px;">
+            <p>{{data.resource_type}}</p>
+        </div>
+        {% endif %}
+        {% endif %}
+        {% if title == 'Gesis' %}
+        {% if data %}
+        <p class='url'>{{data.url}}</p>
+        <li>
+            <h2 class='subtitle'><a href='{{data.url}}' target='_blank'>{{data.title}}</a></h2>
+            <span style="color: gray;"> - {{data.date}}<br>{{data.description}}</span>
+        </li>
+
+        <div style="margin-left:10px;">
+            <p>{{data.resource_type}}</p>
+        </div>
+        {% endif %}
+        {% endif %}
+        {% if title == 'Cordis' %}
+        {% if data %}
+        <p class='url'>{{data.url}}</p>
+        <li>
+            <h2 class='subtitle'><a href='{{data.url}}' target='_blank'>{{data.title}}</a></h2>
+            <span style="color: gray;"> - {{data.date}}<br>{{data.description}}</span>
+        </li>
+
+        <div style="margin-left:10px;">
+            <p>{{data.resource_type}}</p>
+        </div>
+        {% endif %}
+        {% endif %}
+        {% endfor %}
+
+        <!-- collapsible part of results-->
+        {% if datas[5] %}
+            <details><summary> <h1 class='showall'> Show all {{ datas|length }} {{ title }} results for "{{ search_term }}"</h1></summary>
```

Comment:
It is working! Please check 1 and 2:
1. In the 'Show all **N** ... results for QUERY', do you subtract the already displayed five items?, or Is N the same as the length of data?
2. Can you make 'Show more' and 'Show less', instead of 'Show all **N** ... results for QUERY'?

Reply:
It counts all elements of the respective category including those that are already displayed. The description now changes from 'Show all N results' to 'Show less' and back.